lib/chef/resource/openssl_x509_certificate.rb

```diff
@@ -109,13 +109,17 @@ class OpensslX509Certificate < Chef::Resource
       property :ca_key_pass, String,
         description: "The passphrase for CA private key's passphrase."
 
+      property :renew_before_expiry, Integer,
+        description: "The number of days before the expiry. The certificate will be automaticaly renewed when the value is reached.",
+        default: 5
+
       action :create do
         description "Generate a certificate"
 
-        unless ::File.exist? new_resource.path
+        if cert_need_renewall?(new_resource.path, new_resource.renew_before_expiry)
           converge_by("Create #{@new_resource}") do
             file new_resource.path do
-              action :create_if_missing
```

Comment:
There was a design to not overwrite the cert on disk if present that was part of this design originally. You'll want to change that action based on renew_before_expiry being set.

Reply:
I assumed that overwrite the cert on disk only if it is expired should be enought. Ok I will overwrite it only if renew_before_expiry is set and the limit reached.

Reply:
Corrected. Hope it was what needed.